rfqmath/fixed_point.go

```diff
@@ -0,0 +1,116 @@
+package rfqmath
+
+import (
+	"fmt"
+	"math"
+	"strconv"
+)
+
+// FixedPoint is used to represent fixed point arithmetic for currency related
+// calculations. A fixed point consists of a value, and a scale. The value is
+// the integer representation of the number. The scale is used to represent the
+// fractional/decimal component.
+type FixedPoint[T Int[T]] struct {
+	// Value is the value of the FixedPoint integer.
+	Value T
+
+	// Scale is used to represent the fractional component. This always
+	// represents a power of 10. Eg: a scale value of 2 (two decimal
+	// places) maps to a multiplication by 100.
+	Scale int
+}
```

Comment:
I think the field `Value` should be called `Coefficient`. See https://en.wikipedia.org/wiki/Significand for other terms that might fit. It's easier to describe a fixed-point in the BLIP if this field is not called "value". If we rename it we can use sentences like "the value of the coefficient".

And I think `Scale` should have type `uint8`. Because:
* `int` can vary across systems
* we use `uint8` in wire messages
* I don't think we ever want a negative value of `Scale`.

Reply:
Makes sense, renamed `Value` to `Coefficient` and set the unit of `Scale` to `uint8`.

Reply:
I disagree, calling it `Coefficient` just obfuscates the meaning, and it's a long variable name to boot. The value is that, just a value, and scale helps you interpret it. If anything you could rename `Scale` to `Coefficient` as that's the actual multiplier value. 

`Value` is what ends up going on the wire, when I send you the amount of msats to send, am I sending you a value, or coefficient? Which will be easier for someone that doesn't have as much context as to understand: that the oracle sends a value, or a coefficient?

Reply:
Flipped a coin and the magic conch chose `Coefficient`.